tests/fips/git_fips.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Case 1525281  - FIPS: git
+# Summary: Test git command in fips environment, whether can be successful via ssh or https protocol.
+# Maintainer: Dehai Kong <dhkong@suse.com>
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+
+    my $password = $testapi::password;
+    my $username = $testapi::username;
+    my $email    = "you\@example.com";
+    select_console "root-console";
+
+    # Create a test repo
+    assert_script_run("zypper -n in git-core", 190);
+    assert_script_run("mkdir -p repos/qa1;cd repos/qa1");
+    assert_script_run("git init");
+    assert_script_run("echo \"SUSE Test\" > README");
+    assert_script_run("git config --global user.email \"$email\"");
+    assert_script_run("git config --global user.name \"$username\"");
+    assert_script_run("git add README;git commit -m \"Initial commit\"");
+
+    # Clone to a bare git repo
+    assert_script_run("cd ~/repos;git clone --bare qa1 qa0");
+
+    # Clone repo via ssh
+    type_string("clear\n");
+    script_run("git clone ssh://localhost:/root/repos/qa0 qa2 | tee /dev/$serialdev", 0);
+    if (assert_screen("input-yes", 60)) {
```

Comment:
assert_screen with if doesn't make sense. assert_screen returns the match or dies. There is no else